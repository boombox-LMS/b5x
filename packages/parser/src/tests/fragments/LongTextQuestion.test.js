const {
  exampleLongTextQuestionMarkup,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/LongTextQuestion");
const { RawFragmentSchema } = require("@b5x/types");
const { z } = require("zod");

describe("The LongTextQuestion fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleLongTextQuestionMarkup
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed long-text-question fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed long-text-question fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed long-text-question fragment uses the expected data schema", () => {
    const validator = () => {
      z.object({}).parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
