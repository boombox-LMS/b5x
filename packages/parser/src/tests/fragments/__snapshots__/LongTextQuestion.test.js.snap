// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LongTextQuestion fragment builds as expected The parsed long-text-question fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "LongTextQuestion",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "74e5050636f4cd254100c7f855897db1",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
